### PR TITLE
Fix SQLAlchemy and typing_extensions version incompatibility

### PR DESCRIPTION
**Pull request by serena-ruan**
<details><summary>&#x1F6E0 DevTools &#x1F6E0</summary>
<p>

[Open in GitHub Codespaces](https://codespaces.new/serena-ruan/mlflow/pull/10772?quickstart=1)

#### Install mlflow from this PR

```
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/10772/merge
```

#### Checkout with GitHub CLI

```
gh pr checkout 10772
```

</p>
</details>

### Related Issues/PRs

Fix `cannot import name 'TypeAliasType' from 'typing_extensions'` error such as https://github.com/mlflow/mlflow/actions/runs/7393085859/job/20112501588?pr=10741

### How is this PR tested?

- [x] Existing unit/integration tests
- [ ] New unit/integration tests
- [ ] Manual tests

### Does this PR require documentation update?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: requirements/test-requirements.txt [hunk 1]
```diff
@@ -43,4 +43,5 @@ databricks-sdk
 # TODO: Unpin once we support openai 1.0
 openai<1.0
 # Required for showing pytest stats
-psutil
\ No newline at end of file
+psutil
+typing_extensions>=4.6.0
```
Comment (WeichenXu123) [inline requirements/test-requirements.txt:47]: Q: shall we update core-requirement.txt file ?

Comment (WeichenXu123) [reply requirements/test-requirements.txt:47]: discussion:

core-requirement.txt has this line `sqlalchemy<3,>=1.4.0`,

typing_extensions>=4.6.0 should be set as dependency requirement of `sqlalchemy` package.

shall we ask `sqlalchemy` team to address this first, and for now we exclude `2.0.25` version in MLflow core-requirements for now ?

Comment (BenWilson2) [reply requirements/test-requirements.txt:47]: It's updated in their master branch and will be released in 2.1 in the next few weeks. Of note on that release as well: they're dropping python 3.8 support in 2.1. 
Based on this, I think this PR is fine for now as the dependency update is only going to be applied to the next new version which we won't be able to use until we drop support for Python 3.8

**Review by BenWilson2 — APPROVED** [review]
LGTM!

**Pull request merged**